src/CoSELayout.js

```diff
@@ -1439,13 +1560,26 @@ CoSELayout.prototype.tileNodes = function (nodes, minWidth) {
     centerY: 0
   };
 
-  // Sort the nodes in ascending order of their areas
+  if (tilingCompareBy) {
+    organization.idealRowWidth = this.calcIdealRowWidth(nodes, favorHorizontalDim);
+  }
+
+  var getNodeArea = function( n ) {
+    return n.rect.width * n.rect.height;
+  };
+
+  var areaCompareFcn = function(n1, n2){
+    return getNodeArea( n2 ) - getNodeArea( n1 );
+  };
+
+  // Sort the nodes in descending order of their areas
   nodes.sort(function (n1, n2) {
-    if (n1.rect.width * n1.rect.height > n2.rect.width * n2.rect.height)
-      return -1;
-    if (n1.rect.width * n1.rect.height < n2.rect.width * n2.rect.height)
-      return 1;
-    return 0;
+    var cmpBy = areaCompareFcn;
+    if ( organization.idealRowWidth ) {
+      cmpBy = tilingCompareBy;
+    }
+
+    return cmpBy( n1, n2 );
```

Comment:
@metincansiper I think we need to get `n1.id` and `n2.id` as parameters of `cmpBy` function if `organization.idealRowWidth`exists, right? Because, we expect users to use node ids as the parameter of `tilingCompareBy` function.

Reply:
@hasanbalci thanks for noticing it. I fixed it. Looks like I missed it since in my test file nodes were sorted in ascending order even with this bug. Along with that I found another issue in the test/demo file and fixed it as well.